src/app/products/products-mix/create-product-mix/create-product-mix.component.ts

```diff
@@ -0,0 +1,93 @@
+import { Router, ActivatedRoute } from '@angular/router';
+
+/** Custom Services */
+import { ProductsService } from '../../products.service';
+
+/**
+ * Create Product mix component.
+ */
+@Component({
+  selector: 'mifosx-create-product-mix',
+  templateUrl: './create-product-mix.component.html',
+  styleUrls: ['./create-product-mix.component.scss']
+})
+export class CreateProductMixComponent implements OnInit {
+
+  /** Product mix form. */
+  productMixForm: FormGroup;
+  /** Product mix form validity. */
+  productMixFormValidity = false;
+  /** Products mix template data. */
+  productsMixTemplateData: any;
+  /** Products options data. */
+  productsOptionsData: any;
+  /** Products data. */
+  productsData: any;
+
+  /**
+   * Retrieves the Products mix template data from `resolve`.
+   * @param {FormBuilder} formBuilder Form Builder.
+   * @param {ProductsService} productsService Products Service.
+   * @param {ActivatedRoute} route Activated Route.
+   * @param {Router} router Router for navigation.
+   */
+  constructor(private formBuilder: FormBuilder,
+              private productsService: ProductsService,
+              private route: ActivatedRoute,
+              private router: Router) {
+              this.route.data.subscribe(( data: { productsMixTemplate: any }) => {
+      this.productsMixTemplateData = data.productsMixTemplate;
+    });
+  }
+
+  /**
+   * Creates the product mix form and conditional controls of the product mix form.
+   */
+  ngOnInit() {
+    this.createProductMixForm();
+    this.setConditionalControls();
+  }
+
+  /**
+   * Creates the product mix form
+   */
+  createProductMixForm() {
+    this.productsOptionsData = this.productsMixTemplateData.productOptions;
+    this.productMixForm = this.formBuilder.group({
+      'productId': ['', Validators.required]
+    });
+  }
+
+  /**
+   * Sets the conditional controls of the product mix form.
+   */
+  setConditionalControls() {
+    this.productMixForm.get('productId').valueChanges.subscribe(productId => {
+      this.productsData = undefined;
+      this.productMixFormValidity = false;
+      this.productsService.getProductMixTemplate(productId).subscribe((productMixTemplateData: any) => {
+        const restrictedProductsData = productMixTemplateData.restrictedProducts;
+        this.productsData = [...restrictedProductsData, ...productMixTemplateData.allowedProducts];
+        this.productMixForm.removeControl('restrictedProducts');
+        this.productMixForm.addControl('restrictedProducts', new FormControl([...restrictedProductsData.map((restrictedProduct: any) => restrictedProduct.id)], Validators.required));
+        this.productMixFormValidity = true;
+        }
+      );
+    });
+  }
+
+  /**
+   * Submits the product mix form and creates product mix,
+   * if successful redirects to products mix.
+   */
+  submit() {
+    const productMix = { 'restrictedProducts': this.productMixForm.value.restrictedProducts };
```

Comment:
```suggestion
    const productMix = { 
      restrictedProducts: this.productMixForm.value.restrictedProducts 
    };
```
Should work without the quotes.